### PR TITLE
GH-1574 better target node support

### PR DESCRIPTION
**Pull request by hmottestad**
GitHub issue resolved: #1574  

Briefly describe the changes proposed in this PR:

* validates all sh:targetNodes, regardless of if the targets exist or not
* fixes som logging issues
*

### CODE REVIEW

File: core/sail/shacl/src/main/java/org/eclipse/rdf4j/sail/shacl/AST/ShaclProperties.java [hunk 1]
```diff
@@ -48,7 +50,7 @@
 	String flags = "";
 
 	Set<Resource> targetClass = new HashSet<>(0);
-	Set<Value> targetNode = new HashSet<>(0);
+	TreeSet<Value> targetNode = new TreeSet<>(new ValueComparator());
```
Comment (hmottestad) [inline core/sail/shacl/src/main/java/org/eclipse/rdf4j/sail/shacl/AST/ShaclProperties.java]: Create a test that exemplifies this.

**Comment by jeenbroekstra** [comment]
@hmottestad, could I ask you to use the [revised branch naming conventions](https://rdf4j.org/documentation/developer/workflow/#workflow)  for future branches? Especially once we move back to merge commits this will be convenient, as it makes the merge commit's message easier to read.

File: core/sail/shacl/src/main/java/org/eclipse/rdf4j/sail/shacl/planNodes/ValuesBackedNode.java [hunk 2]
```diff
@@ -0,0 +1,118 @@
+/*******************************************************************************
+ * Copyright (c) 2018 Eclipse RDF4J contributors.
```
Comment (jeenbroekstra) [inline core/sail/shacl/src/main/java/org/eclipse/rdf4j/sail/shacl/planNodes/ValuesBackedNode.java]: That looks like the wrong year for a new file

File: core/sail/shacl/src/test/java/org/eclipse/rdf4j/sail/shacl/OrderingTest.java [hunk 3]
```diff
@@ -0,0 +1,185 @@
+/*******************************************************************************
+ * Copyright (c) 2018 Eclipse RDF4J contributors.
```
Comment (jeenbroekstra) [inline core/sail/shacl/src/test/java/org/eclipse/rdf4j/sail/shacl/OrderingTest.java]: wrong year.

File: core/sail/shacl/src/main/java/org/eclipse/rdf4j/sail/shacl/planNodes/GroupByCount.java [hunk 4]
```diff
@@ -74,7 +75,8 @@ private void calculateNext() {
 
 				}
 
-				List<Value> line = Arrays.asList(subject, SimpleValueFactory.getInstance().createLiteral(count));
+				List<Value> line = new ArrayList<>(
+						Arrays.asList(subject, SimpleValueFactory.getInstance().createLiteral(count)));
```
Comment (jeenbroekstra) [inline core/sail/shacl/src/main/java/org/eclipse/rdf4j/sail/shacl/planNodes/GroupByCount.java]: Why are you wrapping this in an Arraylist?

File: core/sail/shacl/src/main/java/org/eclipse/rdf4j/sail/shacl/planNodes/Tuple.java [hunk 5]
```diff
@@ -39,7 +39,7 @@ public Tuple(List<Value> list) {
 	}
 
 	public Tuple(Value... list) {
-		line = Arrays.asList(list);
+		line = new ArrayList<>(Arrays.asList(list));
```
Comment (jeenbroekstra) [inline core/sail/shacl/src/main/java/org/eclipse/rdf4j/sail/shacl/planNodes/Tuple.java]: Same here - why is this needed? It looks redundant.

File: core/sail/shacl/src/test/java/org/eclipse/rdf4j/sail/shacl/OrderingTest.java [hunk 6]
```diff
@@ -0,0 +1,185 @@
...
+			Select select = new Select(connection, "?a <" + RDF.TYPE + "> []", "*");
+			List<Tuple> tuples = new MockConsumePlanNode(select).asList();
+
+			String actual = Arrays.toString(tuples.toArray());
+
+			Collections.sort(tuples);
+
+			String expected = Arrays.toString(tuples.toArray());
+
+			assertEquals(expected, actual);
```
Comment (jeenbroekstra) [inline core/sail/shacl/src/test/java/org/eclipse/rdf4j/sail/shacl/OrderingTest.java]: While these tests (after reading back and forth a bit) seem sensible, I'm a bit lost on what this has to do with issue GH-1574. Did the fix you added change ordering behavior somehow?

File: core/sail/shacl/src/test/java/org/eclipse/rdf4j/sail/shacl/OrderingTest.java (hunk 3 above)
Comment (hmottestad) [reply core/sail/shacl/src/test/java/org/eclipse/rdf4j/sail/shacl/OrderingTest.java]: This is just a renamed file.

File: core/sail/shacl/src/main/java/org/eclipse/rdf4j/sail/shacl/planNodes/ValuesBackedNode.java (hunk 2 above)
Comment (hmottestad) [reply core/sail/shacl/src/main/java/org/eclipse/rdf4j/sail/shacl/planNodes/ValuesBackedNode.java]: Will fix

File: core/sail/shacl/src/main/java/org/eclipse/rdf4j/sail/shacl/planNodes/GroupByCount.java (hunk 4 above)
Comment (hmottestad) [reply core/sail/shacl/src/main/java/org/eclipse/rdf4j/sail/shacl/planNodes/GroupByCount.java]: Arrays.asList is immutable and I need mutability.

File: core/sail/shacl/src/test/java/org/eclipse/rdf4j/sail/shacl/OrderingTest.java (hunk 6 above)
Comment (hmottestad) [reply core/sail/shacl/src/test/java/org/eclipse/rdf4j/sail/shacl/OrderingTest.java]: Ordering test is a renamed file where I’ve moved in some old tests and also added one that checks that the set used to store the target nodes is sorted. Which is an assumption used further down the line.

File: core/sail/shacl/src/main/java/org/eclipse/rdf4j/sail/shacl/planNodes/GroupByCount.java (hunk 4 above)
Comment (hmottestad) [reply core/sail/shacl/src/main/java/org/eclipse/rdf4j/sail/shacl/planNodes/GroupByCount.java]: I've added some comments.
